File: Python_Dsc/SelectionSort.py
# Python program for implementation of Selection Sort

# Function to do selection sort iteratively
def iterativeSelectionSort(arr,n):
    #Traverse through l to len(arr)
    for i in range(0,n-1):
        min = i
        for j in range(i+1,n):
            if(arr[j] < arr[j-1]):
                min = j
        
        temp = arr[i]
        arr[i] = arr[min]
        arr[min] = temp
    return arr      

# Driver code to test above
arr = [12, 11, 13, 5, 6]
print("Input Array is:")
print arr
iterativeSelectionSort(arr, len(arr))
print ("Iterated Selection : Sorted array is:")
print arr


''' Output:
    Input Array is:
    [12, 11, 13, 5, 6]
    Iterated Selection : Sorted array is:
    [5, 6, 11, 12, 13]

'''

File: Java_Dsc/SelectionSort.java
import java.util.*;

class SelectionSort {

    public static int[] doSelectionSort(int[] input, int n){
        
        int i,j,min,temp;

        for(i = 0; i < n-1; i++){
            min = i;
            for( j= i+1; j < n; j++ ){                 // O(n^2) time
                if(input[j] < input[min]){
                    min = j;
                }
            }
            temp = input[i];
            input[i] = input[min];
            input[min] = temp;
        }

        return input;
    }

    public static void main(String[] args){

        int[] arr1 = {10,34,2,56,7,67,88,42};
        System.out.println("Input array is: " + Arrays.toString(arr1));

        int[] arr2 = doSelectionSort(arr1, arr1.length);
        System.out.println("Selection sort is: " + Arrays.toString(arr2));
        
    }
}

/* Output:
 * Input array is: [10, 34, 2, 56, 7, 67, 88, 42]
 * Selection sort is: [2, 7, 10, 34, 42, 56, 67, 88]
  */
